src/DIRAC/FrameworkSystem/Client/AuthManagerData.py

```diff
@@ -0,0 +1,179 @@
+""" This class is located between the client and server part and designed to cache user information requested
+    from the server part.
+"""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import six
+from pprint import pprint
+
+from DIRAC import gLogger, S_OK, S_ERROR
+from DIRAC.Core.Utilities import ThreadSafe, DIRACSingleton
+from DIRAC.Core.Utilities.DictCache import DictCache
+
+__RCSID__ = "$Id$"
+
+
+gCacheProfiles = ThreadSafe.Synchronizer()
+
+
+@six.add_metaclass(DIRACSingleton.DIRACSingleton)
+class AuthManagerData(object):
+  """ Authentication manager
+  """
+  __cacheIdPToIDs = DictCache()
+  # # {
+  # #   <IdP1>: [ <IDs> ],
+  # #   <IdP2>: ...
+  # # }
+
+  __cacheProfiles = DictCache()
+  # # {
+  # #   <ID1>: {
+  # #     DNs: {
+  # #       <DN1>: {
+  # #         ProxyProvider: [ <proxy providers> ],
+  # #         VOMSRoles: [ <VOMSRoles> ],
+  # #         ...
+  # #       },
+  # #       <DN2>: { ... },
+  # #     }
+  # #   },
+  # #   <ID2>: { ... }
+  # # }
+
+  __service = DictCache()
+  # # {
+  # #   crash: bool
+  # # }
+
+  def __init__(self):
+    self.rpc = None
+
+  @gCacheProfiles
+  def getProfiles(self, userID=None):
+    """ Get cache information
+
+        :param str userID: user ID
+
+        :return: dict
+    """
+    if userID:
+      return self.__cacheProfiles.get(userID) or {}
+    return self.__cacheProfiles.getDict()
+
+  @gCacheProfiles
+  def updateProfiles(self, userID, data, time=3600 * 24):
+    """ Add cache information
+
+        :param str userID: user ID
+        :param dict data: ID information data
+        :param int time: lifetime
+    """
+    profileDict = self.__cacheProfiles.get(userID) or {}
+    print('================== CLI DATA updateProfiles ==================')
+    print('User ID: %s' % userID)
+    pprint(profileDict)
+    pprint(data)
+    for k, v in data.items():
+      if v is not None:
+        profileDict[k] = v
+    self.__cacheProfiles.add(userID, time, value=profileDict)
+    ids = self.__cacheIdPToIDs.get(profileDict['Provider'])
+    if isinstance(ids, list) and userID not in ids:
+      self.__cacheIdPToIDs.add(profileDict['Provider'], time, userID)
+
+  def __getRPC(self):
+    """ Get RPC
+    """
+    if not self.rpc:
+      from DIRAC.Core.Base.Client import Client
+      self.rpc = Client()._getRPC(url="Framework/AuthManager", timeout=10)
+    return self.rpc
```

Comment:
To avoid additional imports that are available in AuthManagerClient. The problem is that AuthManagerData is used by the Registry, and it in turn can be imported anywhere. Therefore, to avoid possible looping, AuthManagerData imports the minimum required.